pkg/platform/api/api.go

```diff
@@ -0,0 +1,201 @@
+)
+
+type commandStatus = string
+
+const (
+	Success commandStatus = "Success"
+	Failed  commandStatus = "Failed"
+	Unknown commandStatus = "Unknown"
+)
+
+type CommandResult struct {
+	CmdType    updateCommand `json:"cmd_type"`
+	CmdStatus  commandStatus `json:"cmd_status"`
+	Timestamp  string        `json:"timestamp"`
+	ExitStatus *int32        `json:"exit_status"`
+	Stderr     *string       `json:"stderr"`
+}
+
+type UpdateStatus struct {
+	UpdateState       updateState    `json:"update_state"`
+	AvailableUpdates  []string       `json:"available_updates"`
+	ChosenUpdate      *UpdateImage   `json:"chosen_update"`
+	ActivePartition   *StagedImage   `json:"active_partition"`
+	StagingPartition  *StagedImage   `json:"staging_partition"`
+	MostRecentCommand *CommandResult `json:"most_recent_command"`
+}
+
+type APIClient struct {
+	log logging.Logger
+	httpClient *http.Client
+}
+
+func NewAPIClient() *APIClient {
+	return &APIClient {log: logging.New("api"), httpClient: &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				dialer := net.Dialer{}
+				return dialer.DialContext(ctx, "unix", bottlerocketAPISock)
+			},
+		},
+		// Set a 10 second timeout for all requests
+		Timeout: 10 * time.Second,
+	},
+	}
+}
+
+// makeAPIRequest makes an http request to the Bottlerocket API and returns the response
+// We only retry the request if the response was a 423 Locked (indicates that the update API is not ready)
+func(c *APIClient) makeAPIRequest(method string, path string) (*http.Response, error) {
+	var response *http.Response
+	var attempts = 0
+	// Retry up to 5 times in case the Update API is busy; Waiting 10 seconds between each attempt.
+	for ; attempts < 5; attempts++ {
+		var err error
+		c.log.Debugf("making '%s' request to '%s' API endpoint", method, path)
+		switch method {
+		case "GET":
+			response, err = c.httpClient.Get("http://unix" + path)
+		case "POST":
+			response, err = c.httpClient.Post("http://unix" + path, "text/plain", http.NoBody)
+		default:
+			err = errors.Errorf("unsupported method %s", method)
+		}
+		if err != nil {
+			return nil, errors.Errorf("error when trying to '%s' '%s'", method, path)
+		}
+		if response.StatusCode >= 200 && response.StatusCode < 300 {
+			// Response OK
+			break
+		} else if response.StatusCode == 423 {
+			if attempts < 4 {
+				c.log.Info("API server busy, retrying in 10 seconds ...")
+				// Retry after ten seconds if we get a 423 Locked response (update API busy)
+				time.Sleep(10 * time.Second)
+			}
+		} else {
+			return response, errors.Errorf("bad http response when trying to '%s' '%s': status code %d", method, path, response.StatusCode)
+		}
+	}
+	if attempts == 5 {
+		return nil, errors.New("update API unavailable: retries exhausted")
+	}
+	return response, nil
+}
+
+
+// GetUpdateStatus returns the update status from the update API
+func(c *APIClient) GetUpdateStatus() (*UpdateStatus, error) {
```

Comment:
nit: can you make sure to run `gofmt`?  This is formatted in a non-standard way right now.

```suggestion
func (c *APIClient) GetUpdateStatus() (*UpdateStatus, error) {
```

(we should really add this as an automatic check https://github.com/bottlerocket-os/bottlerocket-update-operator/issues/37)